foundry/entities/user.d.ts

```diff
@@ -145,7 +145,11 @@ declare class User extends Entity<User.Data> {
    * @param fromSlot - An optional origin slot from which the Macro is being shifted
    * @returns A Promise which resolves once the User update is complete
    */
-  assignHotbarMacro(macro: Macro | null, slot?: number | string, { fromSlot }?: { fromSlot?: number }): Promise<User>;
+  assignHotbarMacro(
+    macro: Macro | null,
+    slot?: User.DropData['slot'],
```

Comment:
Please leave this as `string | number`. That's what the method can actually handle (and it's also how it's documented in 0.8.1: https://foundryvtt.com/api/alpha/User.html#assignHotbarMacro)
(this method can also be called in other situations the from drag & drop)